sdk/rust/oak/src/lib.rs

```diff
@@ -465,15 +465,24 @@ pub fn run_event_loop<T: crate::io::Decodable, N: Node<T>>(mut node: N, in_handl
         // transient or not, and then continue or terminate the event loop, respectively.
         match receiver.wait() {
             Err(status) => {
-                error!("error waiting for command: {:?}", status);
                 use crate::OakStatus::*;
                 match status {
-                    ErrTerminated | ErrBadHandle | ErrChannelClosed => {
-                        info!("non-transient error: terminating event loop");
+                    ErrTerminated => {
+                        info!(
+                            "received termination status: {:?}; terminating event loop",
+                            status
+                        );
+                        return;
+                    }
+                    ErrBadHandle | ErrChannelClosed => {
+                        warn!(
+                            "non-transient error waiting for command: {:?}; terminating event loop",
+                            status
+                        );
                         return;
                     }
                     _ => {
-                        info!("(possibly) transient error: continuing event loop");
+                        warn!("(possibly) transient error waiting for command: {:?}; continuing event loop", status);
```

Comment:
I like that now all the logs related this Error are one line (as opposed to error followed by info), makes it very clear that it's about he same thing

Reply:
Also being able to pattern match is nice, it's good to see for this humble JavaScript developer.